### PR TITLE
feat: create a table component gf-22

### PR DESCRIPTION
**Pull request by AnutkaGn**
### Summary

This PR implements the table creation component.

### Changes

- Create a common table component (use React Table)

- For testing purposes place it on the main page with some mock data

### CODE REVIEW

File: apps/frontend/src/libs/components/table/mock-data.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { type ColumnDef } from "@tanstack/react-table";
+
+type Person = {
+	createdAt: string;
+	name: string;
+	permissions: string;
+	projects: string;
+};
+
+const data: Person[] = [
```
Comment (what1s1ove) [inline apps/frontend/src/libs/components/table/mock-data.ts:10]: we do not need any mock data in the PRs

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/table/mock-data.ts (hunk 1 above)
Comment (AnutkaGn) [reply apps/frontend/src/libs/components/table/mock-data.ts:10]: The task says that in order to be able to test, you need to add this component to the main page with mocked data. Should I remove mocked data?

Comment (what1s1ove) [reply apps/frontend/src/libs/components/table/mock-data.ts:10]: i am sorry, i did not notice that comment in the task. then i will resolve the comment 🙏

**Review by what1s1ove — APPROVED** [review]

**Review by yevhenii-hladkov — APPROVED** [review]

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.table-container {
+	align-self: center;
+	width: 1000px;
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/styles.module.css:3]: Would this be responsive with the strict width?

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+.table th,
+.table td {
+	padding: 10px 20px;
+	text-align: left;
+	border-bottom: 1px solid var(--color-border-primary);
+}
+
+.table td {
+	height: 60px;
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/styles.module.css:26]: use min-height instead

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	font-weight: 400;
+}
+
+.table th {
+	height: 43px;
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/styles.module.css:31]: same here, `min-height: 40px` or 45 if it looks better

File: apps/frontend/src/libs/components/table/mock-data.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	},
+	{
+		createdAt: "Jan 26, 2008 12:23",
+		name: "Vel",
+		permissions: "Permission 1, Permission 2",
+		projects: "Project 1, Project 2",
+	},
+];
+
+const columns: ColumnDef<Person>[] = [
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/mock-data.ts:55]:
```suggestion
const mockColumns: ColumnDef<Person>[] = [
```

File: apps/frontend/src/libs/components/table/mock-data.ts (hunk 1 above)
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/mock-data.ts:10]:
```suggestion
const mockData: Person[] = [
```

File: apps/frontend/src/libs/components/app/app.tsx [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
-import { Link, Loader, RouterOutlet } from "~/libs/components/components.js";
+import {
+	Link,
+	Loader,
+	RouterOutlet,
+	Table,
+} from "~/libs/components/components.js";
+import {
+	columns,
+	data,
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/app/app.tsx:10]:
```suggestion
import {
	mockColumns as mockTableColumns,
	mockData as mockTableData,
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Comment by liza-veis** [comment]
Please attach a screenshot on how it looks on ui

**Comment by AnutkaGn** [comment]
Table

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 7]
```diff
@@ -0,0 +1,38 @@
+.table-container {
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table/styles.module.css:1]: Do we need a table container at all in a common component?  It looks like this should be added on the main page, where you place your component. By default, the table will take 100% of it's container and we will add wrappers and containers in the place where we use this table.

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**